src/components/resources.js

```diff
@@ -207,80 +380,495 @@ function Resources(props) {
     setCategory(changedcategoryevent.target.value);
     // console.log(changedcategoryevent.target.value);
   };
+  const appendData = function () {
+    const tempArr = partData.concat(
+      data.slice(partData.length, partData.length + 30)
+    );
+    setPartData(tempArr);
+  };
+
+  const openSharingLink = function (message) {
+    const shareUri = `https://www.addtoany.com/share#url=${encodeURI(
+      'https://www.covid19india.org/essentials'
+    )}&title=${encodeURI(message)}`;
+
+    const h = 500;
+    const w = 500;
+    const left = window.screen.width / 2 - w / 2;
+    const top = window.screen.height / 2 - h / 2;
+    return window.open(
+      shareUri,
+
+      document.title,
+      'toolbar=no, location=no, directories=no, status=no, menubar=no, scrollbars=no, resizable=no, copyhistory=no, width=' +
+        w +
+        ', height=' +
+        h +
+        ', top=' +
+        top +
+        ', left=' +
+        left
+    );
+  };
+
+  const openSharingTray = function () {
+    const message =
+      'Discover nearest coronavirus support and essential service providers such as testing lab centres, accommodation shelters and vegetable vendors at ';
+    if (navigator.share !== undefined) {
+      navigator
+        .share({
+          title: document.title,
+          text: message,
+          url: 'https://www.covid19india.org/essentials',
+        })
+        .then()
+        .catch((error) => console.log(error));
+    } else {
+      openSharingLink(message);
+    }
+  };
   return (
     <div className="Resources">
       <div className="filtersection">
         <div className="filtertitle">
-          <h3>Apply filters and hit search</h3>
+          <h3>Service Before Self</h3>
         </div>
-        <div className="resourcefilters">
-          <div className="resourcefilter">
-            {/* <label for='stateselect1' className='filterlabel'>
-                            State
-                        </label> */}
-            <select
-              id="stateselect1"
-              onChange={changeIndianState}
-              defaultValue=""
+        {!isDesktop && (
```

Comment:
This component is huge, almost 200 lines and the whole component is repeated in an if-else condition. I believe for mobile and for desktop. Can we refactor this code to use only one component and then apply CSS/classes using the condition. It will make the code DRY and more readable.

Reply:
We can write CSS in a separate file and include

Reply:
Yes I know the files have gotten huge. After this release, our first priority is cleaning the code. When I started this, I didn't anticipate this tab would grow this huge. There are lots of developers working on this and we didn't get time to cleanly merge the changes. We will refactor this code surely.